src/bgw/job_stat.c

```diff
@@ -186,30 +189,63 @@ calculate_jitter_percent()
 /* For failures we have standard exponential backoff based on consecutive failures
  * along with a ceiling at schedule_interval * MAX_INTERVALS_BACKOFF */
 static TimestampTz
-calculate_next_start_on_failure(TimestampTz last_finish, int consecutive_failures, BgwJob *job)
+calculate_next_start_on_failure(TimestampTz finish_time, int consecutive_failures, BgwJob *job)
 {
 	float8 jitter = calculate_jitter_percent();
 	/* consecutive failures includes this failure */
 	float8 multiplier = 1 << (consecutive_failures - 1);
+	TimestampTz res;
+	volatile bool res_set = false;
+	TimestampTz last_finish = finish_time;
+	MemoryContext oldctx;
+	if (!IS_VALID_TIMESTAMP(finish_time))
+	{
+		elog(LOG, "calculate_next_start_on_failure, got bad finish_time");
```

Comment:
why/when would this happen?

Would be good to understand and comment on here.

Reply:
it should not happen. I don't see a code path that can lead to it. Bit want to make sure we have valid input.